packages/app/src/cli/models/app/app.ts

```diff
@@ -20,13 +25,85 @@ export const LegacyAppSchema = zod
   .strict()
 
 // adding http or https presence and absence of new lines to url validation
-const validateUrl = (zodType: zod.ZodString) => {
+const validateUrl = (zodType: zod.ZodString, {httpsOnly = false, message = 'Invalid url'} = {}) => {
+  const regex = httpsOnly ? /^(https:\/\/)/ : /^(https?:\/\/)/
   return zodType
     .url()
-    .refine((value) => Boolean(value.match(/^(https?:\/\/)/)), {message: 'Invalid url'})
-    .refine((value) => !value.includes('\n'), {message: 'Invalid url'})
+    .refine((value) => Boolean(value.match(regex)), {message})
+    .refine((value) => !value.includes('\n'), {message})
 }
 
+const ensurePathStartsWithSlash = (arg: unknown) => (typeof arg === 'string' && !arg.startsWith('/') ? `/${arg}` : arg)
+const ensureHttpsOnlyUrl = validateUrl(zod.string(), {
+  httpsOnly: true,
+  message: 'Only https urls are allowed',
+}).refine((url) => !url.endsWith('/'), {message: 'URL can’t end with a forward slash'})
```

Comment:
It's exactly the same as the initial slash from the path. We are removing it in one case, and raising an error in the other... I think we should be consistent at least.

And, taking into account that it's just a formatting issue (because https://www.shopify.com////contact also works), I think it's fine to modify it, even if we finally store something that is not identically the same as the TOML content.